x-pack/plugin/analytics/src/main/java/org/elasticsearch/xpack/analytics/mapper/HistogramFieldMapper.java

```diff
@@ -0,0 +1,459 @@
+                    token = context.parser().nextToken();
+                    //should be an array
+                    if (token != XContentParser.Token.START_ARRAY) {
+                        throw new MapperParsingException("error parsing field ["
+                            + name() + "], expected an [" + XContentParser.Token.START_ARRAY.name() +
+                            "] but got [" + token.name() + "]");
+                    }
+                    values = new ArrayList<>();
+                    token = context.parser().nextToken();
+                    while (token != XContentParser.Token.END_ARRAY) {
+                        values.add(context.parser().doubleValue());
+                        token = context.parser().nextToken();
+                    }
+                } else if (fieldName.equals(COUNTS_FIELD.getPreferredName())) {
+                    token = context.parser().nextToken();
+                    //should be an array
+                    if (token != XContentParser.Token.START_ARRAY) {
+                        throw new MapperParsingException("error parsing field ["
+                            + name() + "], expected an [" + XContentParser.Token.START_ARRAY.name() +
+                            "] but got [" + token.name() + "]");
+                    }
+                    counts = new ArrayList<>();
+                    token = context.parser().nextToken();
+                    while (token != XContentParser.Token.END_ARRAY) {
+                        counts.add(context.parser().intValue());
+                        token = context.parser().nextToken();
+                    }
+                } else {
+                    throw new MapperParsingException("error parsing field [" +
+                        name() + "], with unknown parameter [" + fieldName + "]");
+                }
+                token = context.parser().nextToken();
+            }
+            if (values == null) {
+                throw new MapperParsingException("error parsing field ["
+                    + name() + "], expected field called [" + VALUES_FIELD.getPreferredName() + "]");
+            }
+            if (counts == null) {
+                throw new MapperParsingException("error parsing field ["
+                    + name() + "], expected field called [" + COUNTS_FIELD.getPreferredName() + "]");
+            }
+            if (values.size() != counts.size()) {
+                throw new MapperParsingException("error parsing field ["
+                    + name() + "], expected same length from [" + VALUES_FIELD.getPreferredName() +"] and " +
+                    "[" + COUNTS_FIELD.getPreferredName() +"] but got [" + values.size() + " != " + counts.size() +"]");
+            }
+            if (values.size() == 0) {
+                throw new MapperParsingException("error parsing field ["
+                    + name() + "], arrays for values and counts cannot be empty");
+            }
```

Comment:
I wonder whether we should actually fail it or not.

Reply:
On the other hand, maybe we should require that values come in order and fail if there are duplicate values?

Reply:
Respect empty arrays, yes we can just ignore it instead of failing.

I like he idea of requiring values to be ordered and disallow duplicates. Before implementing I will wait for more input.

Reply:
I updated the code so in case of empty arrays we just ignore it as we do with null values.

In addition we require now that values are provided in increasing order.